### ISSUE TITLE
[ENHANCEMENT] Support HLSL on Xbox

### ISSUE THREAD

**Issue by mzmssg** [microsoft/nnfusion#254]
## Flow
![image](https://user-images.githubusercontent.com/11887940/112420408-1c8c3d00-8d68-11eb-9f78-5b4b8085bcb4.png)

## Items 
Inference
- NNF CLI
  - [ ] NNFusion model provide entry on host memory
  - [ ] Compile NNFusion model into separate DLL
  - [ ] Fix D3D wrapper crash bug
  - [ ] MSBuild NNFusion model
- NNF Python
  - [ ] Support DLL model in Windows
  - [ ] Define data format [Use PyTorch tensor]
- Antares IR
  - [ ] Support more ops 	[ready for MLP]

Training
- NNF CLI
  - [ ] Differential for more ops
  - [ ] More optimizer type
  - [ ] Do not prune backward graph
- NNF Python
  - [ ] Provide interface to read/update/save/load weights
- Antares IR
  - [ ] Support loss ops
  - [ ] Support backward ops

**Comment by nnfbot** [comment]
Thanks for the report @mzmssg! I will look into it ASAP! (I'm a bot).
